python/lsst/the/monster/assemble.py

```diff
@@ -0,0 +1,167 @@
+
+                    # read in spline
+                    colorterm_spline = ColortermSpline.load(colorterm_filename)
+
+                    # apply colorterms to transform to SynthLSST mag
+                    band_1, band_2 = cat_info.get_color_bands(band)
+                    orig_flux = cat_stars[f"decam_{band}_from_{cat_info.name}_flux"]
+                    orig_flux_err = cat_stars[f"decam_{band}_from_{cat_info.name}_fluxErr"]
+                    model_flux = colorterm_spline.apply(
+                        cat_stars[f"decam_{band_1}_from_{cat_info.name}_flux"],
+                        cat_stars[f"decam_{band_2}_from_{cat_info.name}_flux"],
+                        orig_flux,
+                    )
+
+                    # Rescale flux error to keep S/N constant
+                    model_flux_err = model_flux * (orig_flux_err/orig_flux)
+
+                    # Add the fluxes and their errors to the catalog:
+                    cat_stars[f"monster_lsst_{band}_flux"] = model_flux
+                    cat_stars[f"monster_lsst_{band}_fluxErr"] = model_flux_err
+
+                    # Apply selection to only apply transformations within the
+                    # useful color range. (Note that the input DES-system
+                    # catalogs already had their survey-specific cuts applied,
+                    # so these cuts should be for the SynthLSST
+                    # transformations.)
+                    color_range = self.synth_lsst_info.get_color_range(band)
+                    colors = cat_info.get_transformed_mag_colors(cat_stars, band)
+                    selected = (colors >= color_range[0]) & (colors <= color_range[1])
+
+                    flux_not_nan = np.isfinite(cat_stars[f"monster_lsst_{band}_flux"])
+                    flag = selected & flux_not_nan
+                    cat_stars_selected = cat_stars[flag]
+
+                    # Match the transformed catalog to Gaia.
+                    a, b = esutil.numpy_util.match(gaia_stars_all['id'],
+                                                   cat_stars_selected['GaiaDR3_id'])
+
+                    # If the flux measurement is OK, write it to the overall
+                    # Gaia catalog:
+                    flux_col = f"monster_lsst_{band}_flux"
+                    gaia_stars_all[flux_col][a] = cat_stars_selected[flux_col][b]
+                    fluxerr_col = flux_col+'Err'
+                    gaia_stars_all[fluxerr_col][a] = cat_stars_selected[fluxerr_col][b]
+
+                    # Update the flags to denote which survey the flux came
+                    # from:
+                    gaia_stars_all[f"monster_lsst_{band}_source_flag"][a] = cat_info.flag
+
+        # Call this version "monster_v1":
+        write_path_monster = "/sdf/data/rubin/shared/the_monster/sharded_refcats/monster_v1"
+
+        # Output the finished catalog for the shard:
+        if os.path.exists(write_path_monster) is False:
+            os.makedirs(write_path_monster)
```

Comment:
You can do `os.makedirs(write_path_monster, exist_ok=True)` to turn this into a one-liner.